tools/cli/main.js

```diff
@@ -746,6 +746,44 @@ Fiber(function () {
     rawArgs.push(term);
   }
 
+  // Prevent running meteor as root
+
+  var unixPlatforms = {
+    'darwin': true,
+    'linux': true,
+    'freebsd': true
+  };
+  var platform = process.platform;
+
+  if (_.has(unixPlatforms, platform)){
+    // On UNIX platforms.
+    if (process.getgid() === 0 && process.getuid() === 0){
+      if (_.has(rawOptions, '--unsafe-perm')) {
+        // Meteor is running as root and has the --unsafe-perm flag, just notice.
+        Console.info("");
+        Console.info("You have run Meteor as root.",
+        "Your permissions in your app directory will be incorrect if you ever attempt to perform any Meteor tasks as your non-root user.",
+        "You probably didn't want this, but you can fix it by running the following from the root of your project:");
+        Console.info("");
+        Console.info("sudo chown -Rh <username> .meteor/local");
+        Console.info("");
+        // Fix issue when running `sudo meteor --unsafe-perm`, --unsafe-perm: unknown option.
+        delete rawOptions['--unsafe-perm'];
```

Comment:
If the `--unsafe-perm` flag is not deleted the command emit an error because this option is not in the command's available options.